src/Facility.Core/Http/ServiceHttpHandler.cs

```diff
@@ -427,14 +428,19 @@ protected Task<T> AdaptTask<T>(Task<T> task)
 
 		if (routePath.IndexOfOrdinal('{') != -1)
 		{
-			// ReSharper disable once RedundantEnumerableCastCall (needed for .NET Standard 2.0)
-			var names = s_regexPathParameterRegex.Matches(routePath).Cast<Match>().Select(x => x.Groups[1].ToString()).ToList();
-			var regexPattern = Regex.Escape(routePath);
-			foreach (var name in names)
-				regexPattern = regexPattern.ReplaceOrdinal("\\{" + name + "}", "(?'" + name + "'[^/]+)");
-			regexPattern = "^(?:" + regexPattern + ")$";
-			var match = new Regex(regexPattern, RegexOptions.CultureInvariant | RegexOptions.IgnoreCase).Match(requestPath);
-			return match.Success ? names.ToDictionary(name => name, name => Uri.UnescapeDataString(match.Groups[name].ToString())) : null;
+			var regex = s_routePathRegexes.GetOrAdd(routePath, path =>
+			{
+				// ReSharper disable once RedundantEnumerableCastCall (needed for .NET Standard 2.0)
+				var names = s_regexPathParameterRegex.Matches(path).Cast<Match>().Select(x => x.Groups[1].ToString()).ToList();
```

Comment:
`.ToList()` no longer strictly needed, because we only enumerate this once.

Reply:
Clearly `.Aggregate` over the result of `.Matches` should be used to construct `regexPattern` 😆

Reply:
That's what R# wants!